pkg/deploy/configmap_che.go

```diff
@@ -165,29 +123,18 @@ func GetConfigMapData(cr *orgv1.CheCluster) (cheEnv map[string]string) {
 		wsprotocol = "wss"
 		tls = "true"
 	}
-	proxyJavaOpts := ""
-	proxyUser := cr.Spec.Server.ProxyUser
-	proxyPassword := cr.Spec.Server.ProxyPassword
-	proxySecret := cr.Spec.Server.ProxySecret
 
-	nonProxyHosts := cr.Spec.Server.NonProxyHosts
-	if len(nonProxyHosts) < 1 && len(cr.Spec.Server.ProxyURL) > 1 {
-		nonProxyHosts = os.Getenv("KUBERNETES_SERVICE_HOST")
-	} else {
-		nonProxyHosts = nonProxyHosts + "|" + os.Getenv("KUBERNETES_SERVICE_HOST")
-	}
-	if len(cr.Spec.Server.ProxyURL) > 1 {
-		proxyJavaOpts, err = util.GenerateProxyJavaOpts(cr.Spec.Server.ProxyURL, cr.Spec.Server.ProxyPort, nonProxyHosts, proxyUser, proxyPassword, proxySecret, cr.Namespace)
-		if err != nil {
-			logrus.Errorf("Failed to generate java proxy options: %v", err)
+	proxyJavaOpts := ""
+	cheWorkspaceNoProxy := proxy.NoProxy
+	if proxy.HttpProxy != "" {
+		if proxy.NoProxy == "" {
+			cheWorkspaceNoProxy = os.Getenv("KUBERNETES_SERVICE_HOST")
```

Comment:
I'm not 100% sure if the *external* K8S API URL is by default in the cluster-wide proxy settings, but I don't think so.
And in a number of cases it is required to add it in the non-proxy hosts, to allow the Keycloak server to contact the OpenShift ApiServer through this external URL that is configured in Keycloak for the `openshift-v4` provider in case OpenShift OAuth in enabled.

This external URL is already retrieved by the operator by [this function](https://github.com/eclipse/che-operator/blob/master/pkg/util/util.go#L195) to configure Keycloak with Openshift OAuth. So it could be added by default here since it is usually required in typical proxy-based installations.
Of course maybe a boolean field to disable this automatic addition of the external API host in the noproxy field could certainly be useful.